gcpinstancemanager.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	apiv1 "cloud-android-orchestration/api/v1"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"google.golang.org/api/option"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	namePrefix  = "cf-"
+	labelPrefix = "cf-"
+)
+
+// GCP implementation of the instance manager.
+type GCPInstanceManager struct {
+	config      *Config
+	uuidFactory func() string
+	client      *compute.InstancesClient
+}
+
+func NewGCPInstanceManager(
+	config *Config,
+	uuidFactory func() string,
```

Comment:
This forces service users to provide a UUID factor. I think it's better to just create it with a default value here and provide a setter.

Reply:
done